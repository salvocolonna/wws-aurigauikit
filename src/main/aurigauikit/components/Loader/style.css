
	.loader {
    border: 9px solid #f3f3f3;
    border-top: 9px solid #3498db;
    border-right: 9px solid #3498db;
    border-radius: 50%;
    background-color: #fff;
    animation: spin .5s linear infinite;
    border-style: double;
    margin-right: 5px;
    margin-left: -8px;
    margin-top: 0.5px;
    float: left;
    height: auto;
    width: auto;
}
