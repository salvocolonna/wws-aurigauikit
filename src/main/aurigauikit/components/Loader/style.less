@import '../../style/constants.less';

.loader {
  border: 9px solid #f3f3f3;
  border-top: 9px solid @primary-color;
  border-right: 9px solid @primary-color;
  border-radius: 50%;
  background-color: #fff;
  animation: spin 0.5s linear infinite;
  border-style: double;
  margin-right: 5px;
  margin-left: -8px;
  margin-top: 0.5px;
  height: auto;
  width: auto;
  float: right;
  width: 5px;
  height: 5px;
  border: 10px solid #f3f3f3;
  border-top: 10px solid @primary-color;
  border-right: 10px solid @primary-color;
  border-style: double;
  border-radius: 50%;
}

.ant-spin i {
  color: @primary-color;
}
