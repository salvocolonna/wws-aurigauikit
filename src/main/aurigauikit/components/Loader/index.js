import React from "react"
import "./style.css"

export default ({ style }) => (
  <div
    className="loader"
    style={{
      float: "right",
      width: "5px",
      height: "5px",
      border: "10px solid #f3f3f3",
      borderTop: "10px solid #3498db",
      borderRight: "10px solid #3498db",
      borderStyle: "double",
      borderRadius: "50%",
      ...style
    }}
  />
)
