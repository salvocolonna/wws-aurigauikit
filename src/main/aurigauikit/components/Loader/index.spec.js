import React from "react"
import renderer from "react-test-renderer"

import Loader from "./"

describe("Loader", () => {
  it("displays as expected", () => {
    const component = renderer.create(<Loader />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("overrides and adds style attributes", () => {
    const component = renderer.create(
      <Loader style={{ float: "left", transform: "translateY(-50px)" }} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
