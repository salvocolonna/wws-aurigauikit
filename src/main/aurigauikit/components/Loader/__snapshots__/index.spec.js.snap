// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader displays as expected 1`] = `
<div
  className="loader"
  style={
    Object {
      "border": "10px solid #f3f3f3",
      "borderRadius": "50%",
      "borderRight": "10px solid #3498db",
      "borderStyle": "double",
      "borderTop": "10px solid #3498db",
      "float": "right",
      "height": "5px",
      "width": "5px",
    }
  }
/>
`;

exports[`Loader overrides and adds style attributes 1`] = `
<div
  className="loader"
  style={
    Object {
      "border": "10px solid #f3f3f3",
      "borderRadius": "50%",
      "borderRight": "10px solid #3498db",
      "borderStyle": "double",
      "borderTop": "10px solid #3498db",
      "float": "left",
      "height": "5px",
      "transform": "translateY(-50px)",
      "width": "5px",
    }
  }
/>
`;
